rotkehlchen/chain/aggregator.py

```diff
@@ -840,6 +840,10 @@ def _update_balances_after_token_query(
     ) -> None:
         # Update the per account token balance and usd value
         for account, token_balances in balance_result.items():
+            if dsr_proxy_append is False:  # reset all account balances
+                for asset in list(balances[account].assets.keys()):
+                    del balances[account].assets[asset]
```

Comment:
I believe this needs rethinking. You are deleting balacnes way too late in the process but I don't see any explanation for why we have duplicated balances. There has to be two logics that introduce the same balances twice for some reason.

The balances get empty here https://github.com/rotki/rotki/blob/8b1bfb8636391e6f52fbfc2e929aa3226bc8c216/rotkehlchen/chain/aggregator.py#L930 because this logic is executed first. Also I believe there is a bug because if there was balances before and after we have `balance == ZERO` then we don't clean it. But that is a different thing.


So the question that has not been addressed is where do we have double balances. If I'm not mistaken we clean the balances at the beginning in the pointed code and something later duplicates it. Have you checked if https://github.com/rotki/rotki/blob/8b1bfb8636391e6f52fbfc2e929aa3226bc8c216/rotkehlchen/chain/ethereum/defi/zerionsdk.py#L292 is adding the duplicated balances? What two sources are adding the same balance.

Please check if indeed the balances are not empty at the beginning of the query and at what point you insert the balances and then the duplicates.

This doesn't look like solving the issue because you are at one point of the execution removing balances when it should happen at the beginning of the balance query in any case.

Reply:
You're right, but the bug is still the balances not getting cleared, my reason was just wrong

> The balances get empty here
> 
> https://github.com/rotki/rotki/blob/8b1bfb8636391e6f52fbfc2e929aa3226bc8c216/rotkehlchen/chain/aggregator.py#L930


The balances don't get cleared because the user's balance is 0, i pushed an updtae for this
> because this logic is executed first. Also I believe there is a bug because if there was balances before and after we have `balance == ZERO` then we don't clean it. But that is a different thing.
> So the question that has not been addressed is where do we have double balances. If I'm not mistaken we clean the balances at the beginning in the pointed code and something later duplicates it. Have you checked if


There're some tests failing, because they probably didn't count on `balance == ZERO`, I'm taking a look at all that fail after this run